### ISSUE TITLE
Hide the "spill" bucket from categorical histograms behind an advanced setting

### ISSUE THREAD

**Issue by Dexterp37** [mozilla/telemetry-dashboard#570]
The Measurement Dashboard is showing, for categorical histograms, an additional bucket: the *spill* bucket (see the [example](https://mzl.la/2JxNimJ)).

This bucket is not present on the client side, but it's added [by the aggregator](https://github.com/mozilla/python_moztelemetry/blob/a38e07ebd38a00fadf64b9e068825665dced8849/moztelemetry/histogram.py#L147) as a kind of catch-all / overflow bucket.

However, Firefox simply [discards accumulations](https://searchfox.org/mozilla-central/rev/c296d5b2391c8b37374b118180b64cca66c0aa16/toolkit/components/telemetry/TelemetryHistogram.cpp#2288-2291) for unknown labels.

With that said, I think we should either document the meaning of the *spill* bucket on the dashboard (e.g. tooltip?) so that users are not surprised by it, or remove it. I personally lean toward the second option, unless there's a good reason not to :)

**Comment by fbertsch** [comment]
If the bucket isn't adding any value, it makes sense to not show it. I'm hesitant to make the data unavailable though, because (IIRC) telemetry could accumulate on those buckets if the call is outside of `AccumulateCategorical` and instead uses `Accumulate`. Obviously there is also the case of simply erroneous data.

Maybe we could have an advanced setting to enable showing the spill bucket?

**Comment by Dexterp37** [comment]
> Maybe we could have an advanced setting to enable showing the spill bucket?

Good point on calls outside of the categorical API: we don't forbid that, though it's a weird usage.
I like your proposal, +1 on that!

**Comment by fbertsch** [comment]
Great! Let's plan on that then.

**Comment by ab0092** [comment]
@fbertsch i have looked into the UI. Is it the spill section under success and failure that is to be made visible based on an advanced setting?? Some more details on this issue would be really helpful as i would like to work on this one.

**Comment by chutten** [comment]
Ah yes, this one.

The idea here is that some probes (categorical ones, to be precise) come with one final bucket labelled "spill" that wouldn't really be that useful to people looking at the data on the measurement dashboard. The proposal is that we hide the "spill" bucket on the distribution and evolution views and provide an "Advanced Setting" to allow it to be re-shown just in case someone who knows what they're doing wants to take a look.

Does that help, @ab0092 ?

**Comment by ab0092** [comment]
From what i understood, when the data is being shown in the measurement dashboard in table mode its shows three labels **success**, **failure** and **spill**. So the spill label is the one that is to be made visible only on user selection that is to be part of the advanced settings option. Is my understanding correct ??

**Comment by chutten** [comment]
Correct. You can see it with different labels here: https://mzl.la/2BRYHiZ

And you can see it in chart form here: https://mzl.la/2BT7jWo

And you can see it in evolution view here: https://mzl.la/2NpkNuo

**Comment by ab0092** [comment]
Just one question, for the evo dashboard the spill option is already a part of the optional aggregates list so the user already has a option to not view it, so it required in that context also ??

**Comment by chutten** [comment]
I think a complete solution would hide it from the dropdown list unless the advanced setting were checked.

**Comment by ab0092** [comment]
Agreed. So i will start working on this and provide an update once it is ready. Thanks for the explanation.. :-)

**Comment by ab0092** [comment]
@chutten i have done the required changes and also made the PR for it. Please have a look and let me know if anything more is to added. You can view changes here **https://ab0092.github.io/telemetry-dashboard/**. And one more input i require from your end is the label of the advance setting to include spill. I have provided a label as per my judgement but i feel it requires changing it. Thanks..
